lib/player/player-service.ts

```diff
@@ -13,49 +13,74 @@ export namespace playerService {
     minStartingBid: number;
     averageBuyNow: number;
     averageStartingBid: number;
-    samplecount: number;
+    sampleCount: number;
   }
+
   export async function getMarketPrice(
     resourceId: number
   ): Promise<MarketPrice> {
+    let invalidSamples = 0;
     const auctions = await getAuctions(resourceId);
-
     const price: MarketPrice = {
       minBuyNow: Number.MAX_VALUE,
       minStartingBid: Number.MAX_VALUE,
       averageBuyNow: 0,
       averageStartingBid: 0,
-      samplecount: 0
+      sampleCount: 0
     };
 
     auctions.forEach(a => {
-      if (a.itemData.resourceId != resourceId) return;
+      if (a.itemData.resourceId != resourceId) {
+        invalidSamples++;
+        return;
+      }
+
+      // ignore the "default" 10000 buy-now price
+      if (a.buyNowPrice === 10000) {
+        invalidSamples++;
+        return;
+      }
+
+      if (a.buyNowPrice < price.minBuyNow) {
+        price.minBuyNow = a.buyNowPrice;
+      }
 
-      if (a.buyNowPrice < price.minBuyNow) price.minBuyNow = a.buyNowPrice;
-      if (a.startingBid < price.minStartingBid)
+      if (a.startingBid < price.minStartingBid) {
         price.minStartingBid = a.startingBid;
+      }
+
       price.averageBuyNow += a.buyNowPrice;
       price.averageStartingBid += a.startingBid;
-      price.samplecount++;
+      price.sampleCount++;
     });
-    price.averageBuyNow /= price.samplecount;
-    price.averageStartingBid /= price.samplecount;
 
-    price.minBuyNow =
-      price.minBuyNow === Number.MAX_VALUE ? 0 : price.minBuyNow;
+    // prevent division by zero
+    if (price.sampleCount > 0) {
+      price.averageBuyNow /= price.sampleCount;
+      price.averageStartingBid /= price.sampleCount;
+    }
 
+    // if there are more invalid samples than valid ones, invalidate the player

```

Comment:
I don't this this is correct. There will be cases where a player will be listed 10 times with buy now price of 10000 and buy now price of 1000 for 5 times. This doesn't make player's samples invalid. His price should return as 1000

Reply:
Sure thing. I guess the other safety checks are enough to ensure a sure price-assumption.
Removed it 👍